### DISCUSSION TITLE
Global state with Nextjs app router

### DISCUSSION THREAD

**Discussion by dwalker93** [pmndrs/zustand#2102] [Q&A]
Normally when I used zustand, I declared global zustand store and then directly called zustand hooks in client components in nextjs app router. But after watch this Jack Herrington's [Tutorial](https://www.pronextjs.dev/tutorials/state-management/zustand-for-state-management/state-management-with-zustand), I learned we can't safely use Zustand hooks that are declared globally. We need to pass zustand hooks via react context api to prevent store collisions from multiple simultaneously request receiving for App router. Are these steps actually neccessary? Actually can be cause any errors that if we use zustand hooks directly? Please guide me.

**Comment by dai-shi** [comment]
It's necessary, if you want to isolate sessions. If you want to share the state across sessions (and users), it can be global. But, it's quite unlikely.

**Reply by dwalker93** [reply]
But then we've to use context also as a provider. So isn't it easy to use only the react conext rather than use zustand with context? Please correct me if I'm wrong

**Reply by dai-shi** [reply]
context is still limited in terms of missing selectors. but, if you find context propagation more useful, nothing stops it.

**Comment by gyanaranjans** [comment]
I don't think its necessary you can use another zustand store instead of putting in the context .

**Reply by dwalker93** [reply]
I don't get you. Can you give me an idea?
